Patch to add Solaris specific paths for mod-ssl.  This patch has not
yet been submitted upstream.

--- puppetlabs-apache-1.8.1/manifests/mod/ssl.pp.orig   2016-03-31 11:38:48.810710927 -0700
+++ puppetlabs-apache-1.8.1/manifests/mod/ssl.pp        2016-03-31 11:40:38.688179788 -0700
@@ -40,12 +40,31 @@
     }
   }

-  $session_cache = $::osfamily ? {
-    'debian'  => "\${APACHE_RUN_DIR}/ssl_scache(512000)",
-    'redhat'  => '/var/cache/mod_ssl/scache(512000)',
-    'freebsd' => '/var/run/ssl_scache(512000)',
-    'gentoo'  => '/var/run/ssl_scache(512000)',
-    'Suse'    => '/var/lib/apache2/ssl_scache(512000)'
+  case $::osfamily {
+    'debian': {
+      $session_cache = "\${APACHE_RUN_DIR}/ssl_scache(512000)"
+    }
+    'redhat': {
+      $session_cache = '/var/cache/mod_ssl/scache(512000)'
+    }
+    'freebsd': {
+      $session_cache = '/var/run/ssl_scache(512000)'
+    }
+    'gentoo': {
+      $session_cache = '/var/run/ssl_scache(512000)'
+    }
+    'Suse': {
+      $session_cache = '/var/lib/apache2/ssl_scache(512000)'
+    }
+    'solaris': {
+      if versioncmp($apache_version, '2.4') == 0 {
+       $session_cache = '/var/run/apache2/2.4/ssl_scache(512000)'
+      } elsif versioncmp($apache_version, '2.2') == 0 {
+       $session_cache = '/var/run/apache2/2.2/ssl_scache(512000)'
+      } else {
+       $session_cache = '/var/run/ssl_scache(512000)'
+      }
+    }
   }

   ::apache::mod { 'ssl':

